### DISCUSSION TITLE
3.2.1に関する不具合や質問はこちら

### DISCUSSION THREAD

**Discussion by dongaba** [dongaba/TVerRec#108] [Issues]
### 3.2.1 Added

- 動画の移動先ディレクトリがジャンクションとなっていて正常にファイル移動ができない場合の救済オプションを追加
- BS局のタグを追加
- 移動先ディレクトリのWindowsジャンクションのサポート (Thanks to [@seijit](https://github.com/seijit))

### 3.2.1 Changed

- 番組名から「《」と「》」に加え「【」と「】」で挟まれた文字も除去するように変更
- 番組名から「《」と「》」、「【」と「】」で挟まれた文字を削除する際に一定の長さ(10文字)がないと除去しないように変更
- エピソード番号が10の倍数になっている際に正しいエピソード番号を極力類推するように変更

### 3.2.1 Fixed

- GUI実行時にコマンドプロンプトに`True`と出力され続ける不具合を修正
- GUI設定画面表示時にエラーが出る不具合を修正

## What's Changed
* merge from master by @dongaba in https://github.com/dongaba/TVerRec/pull/99
* merge from master by @dongaba in https://github.com/dongaba/TVerRec/pull/100
* Fix minor issues by @dongaba in https://github.com/dongaba/TVerRec/pull/106
* Beta to Master by @dongaba in https://github.com/dongaba/TVerRec/pull/107

**Full Changelog**: https://github.com/dongaba/TVerRec/compare/v3.2.0...v3.2.1

<hr /><em>This discussion was created from the release <a href='https://github.com/dongaba/TVerRec/releases/tag/v3.2.1'>3.2.1</a>.</em>

**Comment by PeaPore** [comment]
![スクリーンショット 2024-11-03 014916](https://github.com/user-attachments/assets/a9d8a110-75a6-4070-a333-4ed248ce7ac6)
![スクリーンショット 2024-11-03 013316](https://github.com/user-attachments/assets/3f66aaa6-54c4-49f8-babc-5d6166b34445)

Linux版だと何も問題がなかったので、家族にも使ってもらうべくWindows版を使い始めて使用感を試しております。
そこで一点問題があり、相談いたします。

ループ処理のバッチ（start_tverrec.cmd）をスタートすると2週目で必ずこける症状が発生しております。
理由は添付の通り、作業フォルダを消す動作が入っている点なのですが、回避方法はありますでしょうか。
設定ファイル操作（GUI版）をすべて見ましたが、該当する箇所を探し出せませんでした。

現在は別にバッチを作って、作業フォルダの存否を確認して、無ければフォルダ作成し、
在っても無くてもstart_tverrec.cmdをcallすることで、とりあえず一周はループさせるという処理で、
手間を省いています。

よろしくお願いいたします。

**Reply by dongaba** [reply]
「作業ディレクトリ」を「ダウンロード先ディレクトリ」配下に配置しているため、「ダウンロード先ディレクトリ」配下の空フォルダを削除する処理で「作業ディレクトリ」が削除されてしまっているものと思われます。
「作業ディレクトリ」を「ダウンロード先ディレクトリ」とは別のフォルダに配置するようにしていただけますでしょうか。(C:\TVerRecTmpなど)

**Reply by PeaPore** [reply]
ありがとうございます。
今さらながら、私自身がマヌケだったことに気づきました（笑）

ご指摘いただいた通り、作業ディレクトリを他のフォルダに移したら問題がなくなり、
ループを行うようになりました。助かりました！

さて続いて、もう1件質問があります。実は、作業ディレクトリを
ダウンロード先ディレクトリにしてしまった理由にもなるのですが、
元々、RAMドライブ（RAMディスク）を設定していて、作業ディレクトリに指定してみたのですが、
これがうまく機能しませんでした。使ったRAMドライブツールはImDisk Toolkit 20240210です。
スクリプトは動くのですが、ダウンロードの時点でスルーしてしまい、一向にダウンロードされません。

youtube-dlの画面・GUIの設定画面、ドライブのアイコンについてキャプチャを撮りました。
ImDiskの場合はうまく機能しない、どのRAMドライブなら機能するなどの知見がありましたら、
ご教示ください。

もし、解決しないようならば、このままSSD上で作業することにいたします。

よろしくお願いいたします。
![スクリーンショット 2024-11-03 213013](https://github.com/user-attachments/assets/fe063824-a766-4f7e-b309-713a67467418)
![スクリーンショット 2024-11-03 214813](https://github.com/user-attachments/assets/042d38d3-da8d-4864-9727-23c6084972ae)
![スクリーンショット 2024-11-03 214013](https://github.com/user-attachments/assets/2e668df9-061b-4db0-83b0-60b87852d447)

**Reply by dongaba** [reply]
私自身もImDiskを使っているので、ImDiskだとダメだと言うことはないと思います。
![image](https://github.com/user-attachments/assets/56f1b116-a15c-4dd4-aa6b-32d50d5f9101)
![image](https://github.com/user-attachments/assets/a2f88fa4-8684-431f-9fd0-b798e4f8f234)
![image](https://github.com/user-attachments/assets/2948e339-5e3f-4bef-9ac6-aff06d8d921b)
![image](https://github.com/user-attachments/assets/70e0b52b-3f85-4ab4-b9af-af162a9f11db)

エラーメッセージを見る限りURLのパースに失敗しているように見えるので、特定のURLでダメなのかとも思ったのですが、当方環境では問題ないようです。
![image](https://github.com/user-attachments/assets/834a1a73-9458-4252-a8f3-fb5f8adcf824)

どうやらyoutube-dlに渡されるパスが`D:\\`であるべきところが`D:`となっているようなので、設定ファイルでの作業フォルダの指定を`D:\`ではなく`D:`で試してみていただけますか?

**Reply by PeaPore** [reply]
早めのご回答、ありがとうございました。
ImDiskで問題ないというのは一つ光明が見えそうです。

さて、試してみますが、こちらの都合で確認に数日かかりそうです。
確認でき次第、ご連絡差し上げます。

**Reply by dongaba** [reply]
3.2.2としてdevブランチに変更を加えてみました。問題なければそのうちmasterブランチにマージします。

**Comment by papagamo** [comment]
macにて使用しています。

番組移動の際に、「頭文字[イニシャル]D First Stage」が
”WARNING: ⚠️ 移動元、または移動先にアクセスできなくなりました - /…移動先ファイルパス…/頭文字[イニシャル]D First Stage"
と、メッセージが出て移動されませんでした。
おそらく、[ ] が悪影響していると思われます。
手動で移動させていますので、特に影響はありませんがご連絡まで。

**Reply by dongaba** [reply]
ご報告ありがとうございます。MacだけでなくWindowsでも同様の問題が出ており、特殊文字を含むPathの扱いに問題がありましたので、3.2.2で修正します。
既にdevブランチの3.2.2では対応済みですので、近日中にmasterブランチにマージします。
